### PR TITLE
Add snapshot deployment support

### PR DESCRIPTION
**Pull request by jingxu97**
Add the snapshotter deployment support and example

### CODE REVIEW

**Comment by davidz627** [comment]
@jingxu97 please rebase and fix conflicts

**Comment by davidz627** [comment]
@jingxu97 sorry integration test is broken right now, I am working on fixing

**Comment by davidz627** [comment]
@jingxu97 could you also add snapshot user guide and link to it in the main README?

**Comment by davidz627** [comment]
Hi @jingxu97 just waiting on this to merge now for Beta,
Could you also remove https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/pull/120/files#diff-46e2ab87e8b542bb21f57e7268513d41R40 these changes, currently they are constants used for integration tests. I will fix it later.

File: deploy/kubernetes/dev/setup-cluster.yaml [hunk 1]
```diff
@@ -1,3 +1,34 @@
...
+  verbs: ["list", "watch", "create", "update", "patch"]
+- apiGroups: ["storage.k8s.io"]
+  resources: ["storageclasses"]
+  verbs: ["watch", "get", "list"]
+- apiGroups: ["admissionregistration.k8s.io"]
+  resources: ["mutatingwebhookconfigurations"]
+  verbs: ["create"]
+- apiGroups: [""]
+  resources: ["endpoints"]
+  verbs: ["list", "watch", "create", "update", "delete", "get"]
```
[the remark this answers is not in the document]
Comment (jingxu97) [reply deploy/kubernetes/dev/setup-cluster.yaml]: I tested it, removed watch. The others seems necessary

File: deploy/kubernetes/dev/controller.yaml [hunk 2]
```diff
@@ -42,7 +55,7 @@ spec:
               mountPath: /csi
         - name: gce-pd-driver
           imagePullPolicy: Always
-          image: gcr.io/dyzz-csi-staging/csi/gce-pd-driver:latest
+          image: gcr.io/example/csi/gce-pd-driver:latest
```
[the remark this answers is not in the document]
Comment (jingxu97) [reply deploy/kubernetes/dev/controller.yaml]: sure, changed it back.

**Comment by jingxu97** [comment]
@davidz627 addressed comments. PTAL

**Comment by davidz627** [comment]
/lgtm
Thanks @jingxu97 !!

**Comment by davidz627** [comment]
Blocked on:
https://github.com/kubernetes-csi/external-provisioner/issues/147
https://github.com/kubernetes-csi/external-provisioner/issues/146

When these two issues are resolved:
* Leader election should be turned off, whatever roles were added to support that can be removed from this PR
* Topology support should be turned off by default

**Comment by jingxu97** [comment]
@davidz627 The PR is updated with new provisioner version. PTAL.

**Pull request merged**